### DISCUSSION TITLE
Angular 14

### DISCUSSION THREAD

**Discussion by Mutondi** [taiga-family/taiga-ui#2673] [Q&A] [answered]
Good Day, 

Has this library been upgraded to Angular 14, we have one project heavily reliant on this and experiencing serious optimization issues due to the library not being IVY compliant. 

Do we have a solution/roadmap it would suck to have to replace this in production

**Comment by loxy** [comment, marked as answer]
This lib is already Ivy compatible. You can use both versions (2 and 3) with Angular 14. But compiling with 2.x is slow because of `ngcc`. But with 3.x this is not a problem anymore. This was released by end of August: https://github.com/Tinkoff/taiga-ui/blob/main/CHANGELOG.md#300-2022-08-30

**Reply by waterplea** [reply]
Adding this this, you can run `ng update @taiga-ui/cdk` to automatically migrate 2.x to 3.x. Let us know if you face any issues with that script.

**Reply by Mutondi** [reply]
This is great guys @loxy and @waterplea , We live to fight another day!
